### PR TITLE
feat: matomo tracker components

### PR DESCRIPTION
**Pull request by melniiv**
## Description
Enable matomo tracker on hobbies site

### Closes

**[DEV-XXX](https://helsinkisolutionoffice.atlassian.net/browse/DEV-XXX):**

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -18,6 +22,12 @@ import EventsConfigProvider from '../common-events/configProvider/ConfigProvider
 import { useEventsApolloClient } from '../domain/clients/eventsApolloClient';
 import useEventsConfig from '../hooks/useEventsConfig';
 
+const matomoInstance = createMatomoInstance({
+  disabled: process.env.NEXT_PUBLIC_MATOMO_ENABLED !== 'true',
+  urlBase: process.env.NEXT_PUBLIC_MATOMO_URL_BASE as string,
+  siteId: Number(process.env.NEXT_PUBLIC_MATOMO_SITE_ID),
+});
```
Comment (nikomakela) [inline src/pages/_app.tsx:29]: a matomo configuration from appConfig should be used instead.

File: src/pages/search/index.tsx [hunk 2]
```diff
@@ -38,20 +39,22 @@ export default function Search() {
...
-        <ApolloProvider client={eventsApolloClient}>
-          <SearchPage
-            SearchComponent={AdvancedSearch}
-            pageTitle={'eventSearch.title'}
-          />
-        </ApolloProvider>
-      }
-      footer={<FooterSection />}
-    />
+    <MatomoWrapper>
```
Comment (nikomakela) [inline src/pages/search/index.tsx:42]: Couldn't the MatomoWrapper be used in _app.tsx instead? No need to use it in every page then.

**Review by nikomakela — CHANGES_REQUESTED** [review]
AppConfig has a getter for matomoConfiguration. The Matomo should be configured in there. Also, couldn't the MatomoWrapper be added in the _app.tsx instead? It would be used in a single file instead of using it in multiple files.

**Comment by nikomakela** [comment]
The example of Matomo tracker has a protocol added in the baseUrl as well. In your code the baseUrl does not include the protocol at the moment and I guess that is the reason why the build process fails. https://www.npmjs.com/package/@datapunt/matomo-tracker-react

File: .github/workflows/staging.yml [hunk 3]
```diff
@@ -58,8 +58,11 @@ jobs:
           DOCKER_BUILD_ARG_NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT: ${{ env.NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT: ${{ env.NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_APP_ORIGIN: ${{ env.NEXT_PUBLIC_APP_ORIGIN }}
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 587
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_ENABLED: 0
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_URL_BASE: '//webanalytics.digiaiiris.com/js/'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 939
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SRC_URL: 'piwik.min.js'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_TRACKER_URL: 'tracker.php'
```
Comment (nikomakela) [inline .github/workflows/staging.yml:64]: are these 2 parameters ever read?

File: .github/workflows/review.yml [hunk 4]
```diff
@@ -43,8 +46,11 @@ jobs:
           DOCKER_BUILD_ARG_NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT: ${{ env.NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT: ${{ env.NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_APP_ORIGIN: ${{ env.CUSTOM_ENVIRONMENT_URL }}
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 587
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_ENABLED: 0
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_URL_BASE: '//webanalytics.digiaiiris.com/js/'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 939
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SRC_URL: 'piwik.min.js'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_TRACKER_URL: 'tracker.php'
```
Comment (nikomakela) [inline .github/workflows/review.yml:52]: are these 2 parameters ever read?

File: src/domain/app/AppConfig.ts [hunk 5]
```diff
@@ -74,21 +74,18 @@ class AppConfig {
   }
 
   static get matomoConfiguration() {
-    const matomoUrlBase = '//webanalytics.digiaiiris.com/js/';
+    const matomoUrlBase = process.env.NEXT_PUBLIC_MATOMO_URL_BASE;
     const matomoEnabled = process.env.NEXT_PUBLIC_MATOMO_ENABLED;
     const matomoSiteId = process.env.NEXT_PUBLIC_MATOMO_SITE_ID;
     const getMatomoUrlPath = (path: string) => `${matomoUrlBase}${path}`;
```
Comment (nikomakela) [inline src/domain/app/AppConfig.ts:80]: Should the environment variable be used instead? If not, then the env-variables related to this should be removed.

**Review by nikomakela — COMMENTED** [review]
There are needless environment variables, or they should be read in the AppConfig too.

File: .github/workflows/production.yml [hunk 6]
```diff
@@ -59,8 +58,11 @@ jobs:
           DOCKER_BUILD_ARG_NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT: ${{ env.NEXT_PUBLIC_EVENTS_GRAPHQL_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT: ${{ env.NEXT_PUBLIC_LINKEDEVENTS_EVENT_ENDPOINT }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_APP_ORIGIN: ${{ env.NEXT_PUBLIC_APP_ORIGIN }}
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 587
-          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_ENABLED: 0
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_URL_BASE: 'https://webanalytics.digiaiiris.com/js/'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SITE_ID: 939
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_SRC_URL: 'piwik.min.js'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_TRACKER_URL: 'tracker.php'
```
Comment (nikomakela) [inline .github/workflows/production.yml:64]: are these 2 parameters ever read?

Comment (melniiv) [reply .github/workflows/production.yml:64]: yes,now they are from config

File: src/pages/search/index.tsx (hunk 2 above)
Comment (melniiv) [reply src/pages/search/index.tsx:42]: should be per nextjs page

File: src/pages/_app.tsx (hunk 1 above)
Comment (melniiv) [reply src/pages/_app.tsx]: yes

**Review by nikomakela — APPROVED** [review]
Good. The new boolean-env-variables could be presented like the rest of them are: with numbers, but it's only a trivial thing.

**Pull request merged**